### ISSUE TITLE
Attention sharing: target K/V should include background pixels, not just the layout component

### ISSUE THREAD

**Issue by engichang1467** [engichang1467/Open-Griffin#1]
## Summary

`shared_attention` builds the target half of `K̂_n` / `V̂_n` from the layout component's pixels only. The paper says it should be the layout component's pixels **plus the background pixels**. Neither the default nor `--no-restrict-target` matches the paper.

## What the paper says

Griffin §4.2, defining the terms of eq. (6) and (7):

> where ⊗ extracts only the masked features, and ⊕ represents concatenation. The terms `K^n_T` and `V^n_T` are the keys and values of the target image, **restricted to the pixels of layout component n and the background pixels**.

and the paragraph immediately after eq. (8):

> Intuitively, in our layout self-attention mechanism, each target pixel attends to other pixels within the same layout component, the corresponding regions in the source image, **as well as the background pixels**. Meanwhile, background pixels attend to all pixels in the target image, as the background generation is mostly driven by text conditioning.

So a foreground query's target context is `component_n ∪ background` — everything except the *other* subjects' regions. That exclusion is what prevents identity leakage between subjects; it was never meant to cut the background off too.

## What the code does

https://github.com/engichang1467/Open-Griffin/blob/main/griffin/attn.py#L83-L84

```python
k_tgt = key[:, :, idx] if restrict_target else key
v_tgt = value[:, :, idx] if restrict_target else value
```

`idx` is the layout component alone (`GriffinAttnProcessor._regions`). So:

- `restrict_target=True` (default) → component only, background dropped. Too narrow.
- `restrict_target=False` (`--no-restrict-target`) → whole target including other subjects. Too wide, reintroduces the cross-subject leakage the mechanism exists to prevent.

The background query path (eq. 8) is correct already: the base `scaled_dot_product_attention(query, key, value)` result is kept for every pixel outside all regions.

## Note on the README

README §"Deliberate gaps" currently frames this as an ambiguity between the equations and the prose:

> The equations write `⊕ K_T` for the full target, but the text says a region attends only to its own component. I implemented the text. `--no-restrict-target` gives you the other reading.

The text does not say "only its own component" — it says component *and background*, in two separate places. Both current options are wrong; the README note should be replaced once this is fixed.

## Possible relevance to the open quality gap

README §Status reports subjects rendering smaller than their boxes allow, with the composition sitting at roughly 65% background. Cutting foreground queries off from all background context is a plausible contributor. Worth re-measuring the mask-area-inside-box numbers after the fix.

## Fix sketch

In `GriffinAttnProcessor._regions`, compute the background index once per call (pixels belonging to no layout), and concatenate it onto each region's target index:

```python
union = torch.zeros_like(layouts[0])
for layout in state.layouts:
    union |= layout
bg_idx = flatten_mask(~union, seq_len)
# per region: idx_with_bg = torch.cat([idx, bg_idx])
```

Note `shared_attention` writes results back with `out[:, :, idx] = ...`, so the *write* index must stay the region alone — only the gathered key/value context widens. Keep those two indices separate.

Test to add alongside `test_shared_attention_isolates_regions`: a region's output must change when the background content changes, and must not change when another region's content changes.
